TEMPLATES.md

```diff
@@ -0,0 +1,101 @@
+
+   | Example: | `registry.svc.ci.openshift.org/ci-op-<input-hash>/stable:${component}` |
+   | - | - |
+
+
+ - `RPM_REPO`
+   The RPM repository URL from which your job's RPMs can be installed.
+   `Template`s depending on this parameter will be processed only after the
+   `rpm` image is built.
+
+   | Example: | `http://rpm-repo-ci-op-<input-hash>.svc.ci.openshift.org` |
+   | - | - |
+
+ - `IMAGE_<component>`
+   The image repository URL from which the `<component>` image can be pulled.
+   If a component name has hyphens, replace them with underscores. For instance,
+   the `$IMAGE_SOME_NAME` parameter would point to the registry URL for `some-name`.
+   Valid `<component>` images are any non-optional images defined in the `images`
+   array. Many parameters of this format can be provided for one `Template`.
+   `Template`s depending on these parameters will be processed only after the
+   `<component>` images are built.
+
+   | Example: | `registry.svc.ci.openshift.org/ci-op-<input-hash>/stable:<component>` |
+   | - | - |
+
+ - `LOCAL_IMAGE_<pipeline-tag>`
+   The image repository URL from which the `<pipeline-tag>` image can be pulled.
+   If a component name has hyphens, replace them with underscores. For instance,
+   the `$LOCAL_IMAGE_SOME_NAME` parameter would point to the registry URL for
+   `some-name`. Valid `<pipeline-tag>` images are `src`, `bin`, `test-bin`, or
+   `rpms`. Many parameters of this format can be provided for one `Template`.
+   `Template`s depending on these parameters will be processed only after the
+   `<pipeline-tag>` images are built.
+
+   | Example: | `registry.svc.ci.openshift.org/ci-op-<input-hash>/pipeline:<pipeline-tag>` |
+   | - | - |
+
+ - `JOB_NAME`
+   The job name as provided by Prow in the `$JOB_SPEC`.
+
+ - `JOB_NAME_SAFE`
+   The `$JOB_NAME` transformed to be a valid Kubernetes object identifier.
+
+ - `JOB_NAME_HASH`
+   A short hash of the `$JOB_NAME`. Append this to the `$NAMESPACE` to create an
+   identifier that is unique across all jobs for the same inputs. Note that this
+   identifier will not be unique across multiple builds of the same job for the
+   same inputs.
+
+## Expected Output From `Template`s
+
+The CI Operator attempts to determine the result of the `Template`'s execution
+as well as assembling output artifacts.
+
+### `Template` Execution Outcome
+
+A `Template` definition is allowed to contain any number of `Pod`s. A `Template`
+is considered successful if every `Container` in every `Pod` in the `Template`
+definition starts at least once and terminates with a `0` exit code.
+
+### `Template` Artifact Output
+
+The CI Operator will add a `Container` named `artifacts` to any `Pod` defined in
+the `Template` that exposes a volume named `artifacts`. All files that are added
+to the volume will be uploaded to the appropriate GCS bucket for the job build
+that executed the `Template`. Failures to retrieve or upload artifacts will not
+impact the overall result of the job.
```

Comment:
It's worth pointing out that this only happens when `--artifacts-dir` is set.

Interestingly, the first sentence is true even without the flag. I wonder why that is.

Reply:
/shrug